internal/data/assets.go

```diff
@@ -77,20 +102,41 @@ func (a *AssetModel) GetByCodeAndIssuer(ctx context.Context, code, issuer string
 	return &asset, nil
 }
 
+// GetByWalletID returns all assets associated with a wallet.
+func (a *AssetModel) GetByWalletID(ctx context.Context, walletID string) ([]Asset, error) {
+	assets := []Asset{}
+	query := `
+		SELECT 
+			a.*
+		FROM 
+			assets a
+		JOIN 
+			wallets_assets wa ON a.id = wa.asset_id
+		WHERE 
+		    deleted_at IS NULL
+			AND wa.wallet_id = $1
```

Comment:
The indentation here is a bit odd.

Reply:
fixed in c3b11e00d64721769e98ca0c497ab056b78e1139